### PR TITLE
overhaul-2

### PR DESCRIPTION
**Pull request by Wilson-Flores**

### CODE REVIEW

File: Bus.cpp [hunk 1]
```diff
@@ -4,86 +4,45 @@
...
 void Bus::rom_write(uint16_t address, uint8_t data) {
-    rom[address] = data;
+    rom_[address] = data;
 }
 
 
 uint8_t Bus::rom_read(uint16_t address) {
-    return rom[address];
+    return rom_[address];
 }
```
Comment (Phantim97) [inline Bus.cpp]: if your parameters are not being modified they should be const

File: instructions/Add.cpp [hunk 2]
```diff
@@ -1,426 +1,426 @@
 #include "../z80.hpp"
 
 void z80cpu::ADD_implied_register() {
-    t_state_cycles = 4;
+    t_state_cycles_ = 4;
```
Comment (Phantim97) [inline instructions/Add.cpp:4]: what is the purpose of state_cycles_?

File: test-cases/test_inc.cpp [hunk 3]
```diff
@@ -1,66 +1,66 @@
...
-//        }
-//
-//        ASSERT_EQ(compare_flag_values, false);
-//    }
-//}
+#include "test_main.hpp"
+
+
+TEST_F(IncTest, INC_implied_register_test){
+    std::vector<int> expected_flag_values = {
```
Comment (Phantim97) [inline test-cases/test_inc.cpp]: Minor nit but this can just be a constexpr std::array

File: instructions/Xor.cpp [hunk 4]
```diff
@@ -2,139 +2,139 @@
...
     // N is reset
     set_flag(ADD_SUB_FLAG, false);
     // C is reset
     set_flag(CARRY_FLAG, false);
 
     //X & Y Flags are copies bit 3 & 5 of the accumulator
-    set_flag(X_FLAG, accumulator & 0x08);
-    set_flag(Y_FLAG, accumulator & 0x20);
+    set_flag(X_FLAG, accumulator_ & 0x08);
+    set_flag(Y_FLAG, accumulator_ & 0x20);
```
Comment (Phantim97) [inline instructions/Xor.cpp:82]: maybe a constexpr variable for the 0x08, 0x20 and 0x80 to prevent magic numbers

File: test-cases/test_bit.cpp [hunk 5]
```diff
@@ -1,256 +1,106 @@
...
\ No newline at end of file
+#include "test_main.hpp"
+
+
+TEST_F(BitTest, BIT_implied_test){
+    std::vector<uint8_t> expected_flag_values_00 = {0x00, 0x54, 0x54, 0x54, 0x54, 0x54, 0x54, 0x54, 0x54};
+    std::vector<uint8_t> expected_flag_values_FF = {0x00, 0x38, 0x38, 0x38, 0x38, 0x38, 0x38, 0x38, 0xB8};
+    std::vector<uint8_t> load_value = {0x00, 0xFF};
+
+    std::vector<std::pair<uint8_t, std::vector<uint8_t>>> register_instructions = {
```
Comment (Phantim97) [inline test-cases/test_bit.cpp]: unordered map is probably better here.

**Review by Phantim97 — CHANGES_REQUESTED** [review]
Some initial comments, will review more later

File: instructions/Add.cpp (hunk 2 above)
Comment (Wilson-Flores) [reply instructions/Add.cpp:4]: Nothing currently. Might be useful for a future implementation of instruction delay.

File: instructions/Add.cpp [hunk 6]
```diff
@@ -1,426 +1,426 @@
...
 
     //X & Y Flags are copies bit 3 & 5 of the accumulator
-    set_flag(X_FLAG, accumulator & 0x08);
-    set_flag(Y_FLAG, accumulator & 0x20);
+    set_flag(X_FLAG, accumulator_ & 0x08);
+    set_flag(Y_FLAG, accumulator_ & 0x20);
 }
 
 
 void z80cpu::ADD_implied_immediate() {
```
Comment (Phantim97) [inline instructions/Add.cpp:128]: if the ADD is it's own component vs the ADC maybe they can be nested objects with operations within the z80 class.

File: instructions/Jump.cpp [hunk 7]
```diff
@@ -157,32 +157,32 @@ void z80cpu::JR_cc_relative() {
...
 void z80cpu::JP_implict_iy() {
-    t_state_cycles = 8;
+    t_state_cycles_ = 8;
 
-    program_counter = index_register_y;
+    program_counter_ = index_register_y_;
 }
 
 
 void z80cpu::DJNZ_immediate() {
```
Comment (Phantim97) [inline instructions/Jump.cpp]: Split into individual classes for these, should not all be owned by z80

File: test-cases/test_inc.cpp (hunk 3 above)
Comment (Wilson-Flores) [reply test-cases/test_inc.cpp]: good idea, ill implement with the rest of the test cases as well.

File: test-cases/test_bit.cpp (hunk 5 above)
Comment (Wilson-Flores) [reply test-cases/test_bit.cpp]: an interesting idea, however i wouldnt be able to make the unordered map a constexpr. like the other ill convert it to a constexpr std::array instead.

**Review by Phantim97 — COMMENTED** [review]
Already merged.

**Pull request merged**